src/sgame/sg_cmds.cpp

```diff
@@ -2252,6 +2252,56 @@ static bool FindRoomForClassChangeVertically(
 	return !tr.startsolid && tr.fraction == 1.0f;
 }
 
+static bool FindRoomForClassChangeLaterally(
+		const gentity_t *ent,
+		const vec3_t fromMins, const vec3_t fromMaxs,
+		const vec3_t toMins, const vec3_t toMaxs,
+		vec3_t newOrigin)
+{
+	float max_x = fabs(fabs(toMaxs[0]-toMins[0])
```

Comment:
maxs are always higher than mins so you can remove some `fabs`.

Since player bounding boxes are always centered on the origin, a tighter bound can be achieved by dividing by 2 relative to the current values. (Though a fudge factor beyond the tighest bound would be good since it's floating point, like add 1 at the end.) I think the tighter bound is a good idea because if the full difference between bbox sizes is used, then the new bbox at the start of the trace may not overlap with the player's current bbox. This would make it possible to emerge on the other side of a wall.

Reply:
> Since player bounding boxes are always centered on the origin

Either I misunderstand you, or it is wrong:

In https://github.com/UnvanquishedAssets/res-players_src.dpkdir/blob/master/configs/classes/human_light.model.cfg#L8-L10 we can see that, notably, this model (which can crouch) have origin not centered when standing up, but even more when crouching. IIRC, this is a general rule: height is not centered. But then, you're commenting on a X axis, and this assertion seems true for width and depth, indeed.

Reply:
I've changed the design to me more conservative in order to limit bugs a bit, but it still relies on the assumption you told a bit earlier, for the sake of simplicity.